build.rs

```diff
@@ -87,137 +90,90 @@ fn write_host_rs(mut out: File, triple: Triple) -> io::Result<()> {
-        triple.architecture
-    )?;
-    writeln!(
-        out,
-        "            vendor: Vendor::{},",
-        vendor_display(&triple.vendor)
-    )?;
-    writeln!(
-        out,
-        "            operating_system: OperatingSystem::{:?},",
-        triple.operating_system
-    )?;
-    writeln!(
-        out,
-        "            environment: Environment::{:?},",
-        triple.environment
-    )?;
     writeln!(
         out,
-        "            binary_format: BinaryFormat::{:?},",
-        triple.binary_format
+        r#"
+#[allow(unused_imports)]
+use crate::Aarch64Architecture::*;
+#[allow(unused_imports)]
+use crate::ArmArchitecture::*;
+#[allow(unused_imports)]
+use crate::CustomVendor;
+#[allow(unused_imports)]
+use crate::Mips32Architecture::*;
+#[allow(unused_imports)]
+use crate::Mips64Architecture::*;
+#[allow(unused_imports)]
+use crate::Riscv32Architecture::*;
+#[allow(unused_imports)]
+use crate::Riscv64Architecture::*;
+#[allow(unused_imports)]
+use crate::X86_32Architecture::*;
+
+/// The `Triple` of the current host.
+pub const HOST: Triple = Triple {{
+    architecture: Architecture::{architecture:?},
+    vendor: Vendor::{vendor},
+    operating_system: OperatingSystem::{operating_system:?},
+    environment: Environment::{environment:?},
+    binary_format: BinaryFormat::{binary_format:?},
+}};
+
+impl Architecture {{
+    /// Return the architecture for the current host.
+    pub const fn host() -> Self {{
+        Architecture::{architecture:?}
+    }}
+}}
+
+impl Vendor {{
+    /// Return the vendor for the current host.
+    pub const fn host() -> Self {{
+        Vendor::{vendor}
+    }}
+}}
+
+impl OperatingSystem {{
+    /// Return the operating system for the current host.
+    pub const fn host() -> Self {{
+        OperatingSystem::{operating_system:?}
+    }}
+}}
+
+impl Environment {{
+    /// Return the environment for the current host.
+    pub const fn host() -> Self {{
+        Environment::{environment:?}
+    }}
+}}
+
+impl BinaryFormat {{
+    /// Return the binary format for the current host.
+    pub const fn host() -> Self {{
+        BinaryFormat::{binary_format:?}
+    }}
+}}
+
+impl Triple {{
+    /// Return the triple for the current host.
+    pub const fn host() -> Self {{
+        Self {{
+            architecture: Architecture::{architecture:?},
+            vendor: Vendor::{vendor},
+            operating_system: OperatingSystem::{operating_system:?},
+            environment: Environment::{environment:?},
+            binary_format: BinaryFormat::{binary_format:?},
+        }}
+    }}
+}}"#,
+        architecture = triple.architecture,
+        vendor = vendor_display(&triple.vendor),
+        operating_system = triple.operating_system,
+        environment = triple.environment,
+        binary_format = triple.binary_format,
     )?;
-    writeln!(out, "        }}")?;
-    writeln!(out, "    }}")?;
-    writeln!(out, "}}")?;
 
-    Ok(())
+    out.flush()
```

Comment:
It isn't necessary to `flush()` a `File` because `File`s are unbuffered and their `flush()` doesn't do anything.

Reply:
Thanks, I've reverted this change.